include/openssl/evp.h

```diff
@@ -691,20 +691,23 @@ OPENSSL_EXPORT int EVP_PKEY_keygen_init(EVP_PKEY_CTX *ctx);
 // containing the result. It returns one on success or zero on error.
 OPENSSL_EXPORT int EVP_PKEY_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY **out_pkey);
 
-OPENSSL_EXPORT int EVP_PKEY_encapsulate_init(EVP_PKEY_CTX *ctx,
-                                             const OSSL_PARAM params[]);
+// EVP_PKEY_encapsulate is an operation defined for a KEM (Key Encapsulation
+// Mechanism). The function generates a shared secret and a ciphertext which
+// encapsulates the shared secret using the public key given within |ctx|.
 OPENSSL_EXPORT int EVP_PKEY_encapsulate(EVP_PKEY_CTX *ctx,
-                                        unsigned char *wrapped_key,
-                                        size_t *wrapped_key_len,
-                                        unsigned char *gen_key,
-                                        size_t *gen_key_len);
-OPENSSL_EXPORT int EVP_PKEY_decapsulate_init(EVP_PKEY_CTX *ctx,
-                                             const OSSL_PARAM params[]);
+                                        uint8_t *ciphertext,
+                                        size_t *ciphertext_len,
+                                        uint8_t *shared_secret_key,
```

Comment:
Did you consider naming these `ciphertext_len_out` to make it a bit more clear? Or is already clear that encapsulate will produce a ciphertext and shared_secret_key?